### PR TITLE
Add `LazyList.{memberOf, replace}`

### PR DESCRIPTION
**Pull request by jaschdoc**
Resolves #2033 

- [x] Add `LazyList.memberOf`
- [x] Add `LazyList.prepend`
- [x] Add `LazyList.replace`
- [x] Remove `LazyList.prepend`
- [x] Implement other feedback

### CODE REVIEW

File: main/src/library/LazyList.flix [hunk 1]
```diff
@@ -594,4 +594,29 @@ namespace LazyList {
         case LList(xs)    => flatMapEHelper(f, force xs, k)
     }
 
+    ///
+    /// Returns `true` if and only if `l` contains the element `a`.
+    ///
+    /// Forces elements until `a` is found.
+    ///
+    @Time(length(l)) @Space(1)
```
Comment (magnus-madsen) [inline main/src/library/LazyList.flix:602]: We don't use these annotations here.

File: main/src/library/LazyList.flix [hunk 2]
```diff
@@ -594,4 +594,29 @@ namespace LazyList {
...
+    pub def memberOf(a: a, l: LazyList[a]): Bool with Eq[a] = match l {
+        case ENil         => false
+        case ECons(x, xs) => if (x == a) true else memberOf(a,       xs)
```
Comment (magnus-madsen) [inline main/src/library/LazyList.flix:605]: I think `x == a || memberOf(a, xs)` would be better.

File: main/src/library/LazyList.flix [hunk 3]
```diff
@@ -594,4 +594,29 @@ namespace LazyList {
...
+        case LCons(x, xs) => if (x == a) true else memberOf(a, force xs)
+        case LList(xs)    => memberOf(a, force xs)
+    }
+
+    ///
+    /// Returns `l2` prepended to `l1`. This operation is equivalent to `append(l2, l1)`.
+    ///
+    pub def prepend(l1: LazyList[a], l2: LazyList[a]): LazyList[a] =
```
Comment (magnus-madsen) [inline main/src/library/LazyList.flix:613]: I am not sure why we have both prepend and append. But okay?

File: main/src/library/LazyList.flix (hunk 1 above)
Comment (jaschdoc) [reply main/src/library/LazyList.flix:602]: Ah yeah forgot to remove it :)

File: main/src/library/LazyList.flix (hunk 3 above)
Comment (jaschdoc) [reply main/src/library/LazyList.flix:613]: I put it on the todo list at some point but I'm not sure it totally make sense either. Prepending and appending are equivalent operations with the parameters swapped - something that can be done manually by the user when needed. In short, I think we should remove it.

Comment (magnus-madsen) [reply main/src/library/LazyList.flix:613]: Lets remove it. (and do a quick search to see if there is something like that elsewhere that should be removed too. You can use search in path in IDEA)

**Comment by magnus-madsen** [comment]
Nice! Thanks!

**Pull request merged**